tcms/telemetry/static/telemetry/js/testing/execution-dashboard.js

```diff
@@ -16,23 +16,43 @@ function preProcessData (data, callbackF) {
 
     jsonRPC('TestCase.filter', { pk__in: caseIds }, function (cases) {
         const testerPerCase = {}
+        const componentsPerCase = {}
         cases.forEach(function (element) {
             testerPerCase[element.id] = element.default_tester__username
         })
 
-        jsonRPC('TestRun.filter', { pk__in: runIds }, function (runs) {
-            const testerPerRun = {}
-            runs.forEach(function (element) {
-                testerPerRun[element.id] = element.default_tester__username
+        jsonRPC('Component.filter', { cases__in: cases.map(i => i.id) }, function (components) {
+            components.forEach(function (component) {
+                if (componentsPerCase[component.cases] === undefined) {
+                    componentsPerCase[component.cases] = []
+                }
+                componentsPerCase[component.cases].push(component.name)
             })
 
-            // augment data set with additional info
-            data.forEach(function (element) {
-                element.default_tester__from_case = testerPerCase[element.case]
-                element.default_tester__from_run = testerPerRun[element.run]
+            jsonRPC('TestRun.filter', { pk__in: runIds }, function (runs) {
+                jsonRPC('Product.filter', { pk__in: runs.map(i => i.build__version__product) }, function (products) {
+                    const productPerRun = {}
+                    const productNames = {}
+                    const testerPerRun = {}
+                    products.forEach(function (product) {
+                        productNames[product.id] = product.name
+                    })
```

Comment:
As a side note products are already fetched once before for the Product drop down field so we may start thinking about optimizing this out after everything else is confirmed to work.

Reply:
FTR: this optimization, while possible, isn't going to be straight forward so I'm leaving this out of scope for now at the expense of a secondary request to Product.filter.